core/src/main/scala/kafka/log/Log.scala

```diff
@@ -596,8 +600,8 @@ class Log(@volatile var dir: File,
       if (batchSize > config.maxMessageSize) {
         BrokerTopicStats.getBrokerTopicStats(topicPartition.topic).bytesRejectedRate.mark(records.sizeInBytes)
         BrokerTopicStats.getBrokerAllTopicsStats.bytesRejectedRate.mark(records.sizeInBytes)
-        throw new RecordTooLargeException(s"Message size is $batchSize bytes which exceeds the maximum configured " +
-          s"message size of ${config.maxMessageSize}.")
+        throw new RecordTooLargeException(s"The record batch size is $batchSize bytes which exceeds the maximum configured " +
```

Comment:
This is a timely update. Gwen came by this morning with a good point: Before the new message format changes, when compression was disabled, we would validate the size of each individual message and would throw a `RecordTooLargeException` if any one of them is too large. 

With the new message format we no longer are checking for the size of each individual message on the broker. We still do it on the producer, but it maybe configured differently than the broker. On the broker, we are checking that the entire batch is less than the `maxMessageSize`. This is a breaking change. 

Obviously, we don't have to address it in this patch, but we need to have a discussion and figure out what we do about it before the release.